common/interpreter/invocation.json

```diff
@@ -0,0 +1,114 @@
+{
```

Comment:
afaict, the schema is only used for documentation?   Did we ever consider code generation? While the set of code-generator for JSON doesn't seem as rich as for, say, xml (i know i know ....) or ASN.1 (alas, mostly commercial), there seem to be some choices in json-binding for protobuf, avro  or quicktype.io?  I have to admit i am a fan of such code generators as they guarantee robustness, in particular when refactoring and some you writting tedious (and brittle) boilerplate code. Admittedly, probably something which would have to be done pervasively from the start but just curious ...

PS: if i am correct and this is only for documentation, wouldn't it then also be better to keep all such message definitions together instead of intermingling them with the code?  Or put differently, the purpose might be easily overlooked with the current location?

Reply:
a commit that is in the queue will address this and enforce schema on requests and responses. since the environment is created by the enclave, it is for documentation purposes only (that is, it is for the contract writer). i'm not particularly interested in automating the transition from this schema to a validator. it would take far longer to write that code than we would save by automating it.